pkg/jx/cmd/create_cluster_gke.go

```diff
@@ -550,3 +578,71 @@ func validateClusterName(clustername string) error {
 	}
 	return nil
 }
+
+// enableTenantCluster creates a managed zone which is a sub-domain
+// of a parent domain.
+func enableTenantCluster(tenantServiceURL string, projectID string, zone string) (string, error) {
+
+	log.Infof("Configuring CloudBees Domain for %s project and %s zone", projectID, zone)
+	// Create a TenantClient
+	tCli := tenant.NewTenantClient()
+	var err error
+	domain, err := tCli.GetTenantSubDomain(tenantServiceURL, projectID)
+	if err != nil {
+		return "", errors.Wrap(err, "getting domain from tenant service")
+	}
+	err = validateDomainName(domain)
+	if err != nil {
+		return "", errors.Wrap(err, "domain name failed validation")
+	}
+
+	// Checking whether dns api is enabled
+	err = gke.EnableAPIs(projectID, "dns")
+	if err != nil {
+		return "", errors.Wrap(err, "enabling the dns api")
+	}
+
+	// Create domain if it doesn't exist and return name servers list
+	nameServers, err := createTenantsSubDomainDNSZone(projectID, domain)
+	if err != nil {
+		return "", err
+	}
+
+	log.Infof("%s domain is operating on the following nameservers %v", domain, nameServers)
+	err = tCli.PostTenantZoneNameServers(tenantServiceURL, projectID, domain, zone, nameServers)
+	if err != nil {
+		return "", errors.Wrap(err, "posting the name service list to the tenant service")
+	}
+
+	return domain, nil
+}
+
+// validateDomainName checks for compliance in a supplied domain name
+func validateDomainName(domain string) error {
+	// Check whether the domain is greater than 3 and fewer than 63 characters in length
+	if len(domain) < 3 || len(domain) > 63 {
+		err := fmt.Errorf("domain name %v has fewer than 3 or greater than 63 characters", domain)
+		return err
+	}
+	// Ensure each part of the domain name only contains lower/upper case characters, numbers and dashes
+	if !allowedDomainRegex.MatchString(domain) {
+		err := fmt.Errorf("domain name %v contains invalid characters", domain)
+		return err
+	}
+	return nil
+}
+
+// createTenantsSubDomainDNSZone creates the tenants DNS zone if it doesn't exist
+// and returns the list of name servers for the given domain and project
+func createTenantsSubDomainDNSZone(projectID string, domain string) ([]string, error) {
+	var nameServers = []string{}
+	err := gke.CreateManagedZone(projectID, domain)
+	if err != nil {
+		return []string{}, err
```

Comment:
I would add some context to the error in case it fails, it is more obvious where.